test/test_utils/test_reporting.py

```diff
@@ -210,26 +210,42 @@ def get_marker_arg_value(self, item, function_key, marker_name, default=None):
         else:
             return markers[0].args[0]
 
+    @staticmethod
+    def generate_sm_venvs(venv_path):
+        ctx = Context()
+        EnvBuilder(with_pip=True).create(venv_path)
+        requirements_path = os.path.join(os.path.dirname(venv_path), "requirements.txt")
+        pip_path = os.path.join(venv_path, "bin", "pip")
+        ctx.run(
+            f"{pip_path} install --upgrade pip && {pip_path} install -r {requirements_path}",
+            warn=True,
+        )
+
     def generate_sagemaker_reports(self):
         """
         Append SageMaker data to the report
         """
         ctx = Context()
         git_repo_path = get_repository_local_path()
 
+        venv_paths = []
         for repo in self.SM_REPOS:
             framework, job_type = repo.split(os.sep)
             pytest_framework_path = os.path.join(
                 git_repo_path, "test", "sagemaker_tests", framework, job_type
             )
+            venv_paths.append(os.path.join(pytest_framework_path, f".{repo.replace('/', '-')}"))
+
+        # install venvs in parallel
+        with futures.ThreadPoolExecutor() as executor:
+            executor.map(self.generate_sm_venvs, venv_paths)
+
+        for venv in venv_paths:
+            pytest_framework_path = os.path.dirname(venv)
             with ctx.cd(pytest_framework_path):
-                # We need to install requirements in order to use the SM pytest frameworks
-                venv = os.path.join(pytest_framework_path, f".{repo.replace('/', '-')}")
-                ctx.run(f"virtualenv {venv}")
                 with ctx.prefix(f"source {os.path.join(venv, 'bin', 'activate')}"):
-                    ctx.run("pip install -r requirements.txt", warn=True)
                     # TF inference separates remote/local conftests, and must be handled differently
-                    if framework == "tensorflow" and job_type == "inference":
+                    if "tensorflow" in venv and "inference" in venv:
```

Comment:
Would the new condition be true for hugginface_tensorflow inference images? Since it is not a part of SM_REPOS, I guess this would not have any impact. Just wanted to make sure we make it future proof.